public/app/plugins/panel/logs/LogsPanelEditor.tsx

```diff
@@ -25,8 +55,30 @@ export class LogsPanelEditor extends PureComponent<PanelEditorProps<Options>> {
     onOptionsChange({ ...options, sortOrder: item.value });
   };
 
+  onDataLinksChange = (dataLinks: DataLink[]) => {
+    const { options, onOptionsChange } = this.props;
+    onOptionsChange({ ...options, dataLinks });
+  };
+
+  getSuggestions = async (): Promise<VariableSuggestion[]> => {
```

Comment:
As I added DataLinks here it needs also correct suggestions. As normal DataLinks suggestions does not work (they are based on Metrics context) we need to get the proper suggestions based on what derived fields datasource defines.